product_sticker/models/product_sticker.py

```diff
@@ -91,6 +96,27 @@ def _build_sticker_domain_company(self):
             ]
         )
 
+    @api.model
+    def _build_sticker_domain_model(self, model_names):
+        """Build domain for available models"""
+        if not model_names:
+            return None
+        return expression.OR(
+            [
+                [
+                    (
+                        "available_model_ids",
+                        "in",
+                        self.env["ir.model"]
+                        .sudo()
+                        .search([("model", "in", model_names)])
+                        .ids,
+                    )
+                ],
+                [("available_model_ids", "=", False)],
+            ]
+        )
```

Comment:
IIUC this should be equivalent:

```suggestion
        return [
            "|",
            ("available_model_ids.model", "in", model_names),
            ("available_model_ids", "=", False),
        ]
```